client/blocks/happychat/chat-client/index.jsx

```diff
@@ -91,26 +63,6 @@ export class HappychatClient extends Component {
 	}
 }
 
-const mapState = state => {
-	return {
-		user: getCurrentUser( state ),
-		skills: getSkills( state, getSelectedSiteId( state ) ),
-	};
-};
-
-const mapDispatch = {
-	receiveAccept,
-};
-
-const mergeProps = ( propsFromState, propsFromDispatch, ownProps ) => {
-	return {
-		...ownProps,
-		...propsFromDispatch,
-		...propsFromState,
-		// use component props if we have them, otherwise default to state props
-		user: ownProps.user ? ownProps.user : propsFromState.user,
-		skills: ownProps.skills ? ownProps.skills : propsFromState.skills,
-	};
-};
+const mapState = state => ( { state } );
```

Comment:
Do we need to pass in the whole state object? That way, we lose all the control over how Happychat client uses and accesses the Calypso state. For example, if we decide to change underlying state paths to some data in calypso, we would break happychat client unknowingly. This seems to me like the `add_filter` hooks in wp.org which cause huge issues with backward compatibility as they give too much control over wp to plugins.

Nit: instead of defining a fn, you could just pass that into `connect` directly.

Reply:
I agree that sending the whole state might not have been the most elegant way, I refactored it. However, it was just used in the helper function the state was not actually sent to HappyChat client.